test/legacy-upgrade/mzcompose.py

```diff
@@ -144,9 +145,23 @@ def test_upgrade_from_version(
         )
         with c.override(mz_from):
             c.up("materialized")
+
+            for key, value in mz_from.system_parameter_defaults.items():
+                c.sql(
+                    f"ALTER SYSTEM SET {key} = '{value}'",
+                    port=6877,
+                    user="mz_system",
+                )
     else:
         c.up("materialized")
 
+        for key, value in DEFAULT_SYSTEM_PARAMETERS.items():
```

Comment:
The CI and also my local testdrive run

```bash
bin/mzcompose --find testdrive run default array.td
```

still fail the consistency check, even though I'm now seeing an

```sql
ALTER SYSTEM SET enable_connection_validation_syntax = true
```

corresponding to this call when I run the test locally. However with this line added directly to `array.td` the test passes.

```sql
$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
ALTER SYSTEM SET enable_connection_validation_syntax = true;
```